coderd/workspaceagents.go

```diff
@@ -1908,15 +1900,45 @@ func (api *API) watchWorkspaceAgentMetadata(rw http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	metadataMap := make(map[string]database.WorkspaceAgentMetadatum)
-	for _, datum := range md {
+	log.Debug(ctx, "got initial metadata", "num", len(initialMD))
+
+	metadataMap := make(map[string]database.WorkspaceAgentMetadatum, len(initialMD))
+	for _, datum := range initialMD {
 		metadataMap[datum.Key] = datum
 	}
+	//nolint:ineffassign // Release memory.
+	initialMD = nil
```

Comment:
My understanding here is that we only need the initial slice of metadata from the DB once in order to massage it into a map.
Without this, I assume the initial slice would stay around for the duration of the SSE request?